ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -150,7 +150,43 @@ def gitlab_merge_request_previous_commit_sha(verbose: bool) -> Optional[str]:
     return last_commit
 
 
+def jenkins_previous_commit_sha(verbose: bool) -> Optional[str]:
+    previous_commit = os.getenv("GIT_PREVIOUS_COMMIT")
+    target_branch = os.getenv("CHANGE_TARGET")
+    current_commit = os.getenv("GIT_COMMIT")
+
+    if verbose:
+        click.echo(
+            f"GIT_PREVIOUS_COMMIT: {previous_commit}\n"
+            f"CHANGE_TARGET: {target_branch}\n",
+            err=True,
+        )
+
+    if target_branch:
+        return get_last_commit_sha_of_branch(f"origin/{target_branch}")
+
+    if previous_commit:
```

Comment:
Can you add some logs also here ? In case of a job failure, we want to be able to understand in which case we did end up.